### PR TITLE
json state server link shortener

### PR DESCRIPTION
**Pull request by fcollman**
this provides link shortening capability to neuroglancer through a json state server, which we have currently deployed.  Note the default state server will only presently work at princeton and allen until we get authentication setup.

### CODE REVIEW

File: src/neuroglancer/viewer.ts [hunk 1]
```diff
@@ -600,6 +612,63 @@ export class Viewer extends RefCounted implements ViewerState {
...
+    }
+  }
+
+  postJsonState() {
+    // if jsonStateServer is not present prompt for value and store it in state
+    if (!this.jsonStateServer.value) {
+      console.log('no state server found');
+      this.promptJsonStateServer('no state server found');
+    }
+    ///   let promise = sendHttpJsonPostRequest(openHttpRequest(`${url}/1.0/graph/root`, 'POST'),
```
Comment (nkemnitz) [inline src/neuroglancer/viewer.ts]: Remove unrelated commented code

**Comment by fcollman** [comment]
I've replaced the console logs with status messages and removed the comments.  Go ahead and just squash and merge this when you merge it in.

File: src/neuroglancer/viewer.ts [hunk 2]
```diff
@@ -617,50 +617,55 @@ export class Viewer extends RefCounted implements ViewerState {
...
             this.state.restoreState(response);
-          })
-          .catch(() => {
-            console.log('failed to load from: ' + json_url);
-          });
-    }
+          }),
+              {
+                initialMessage: `Retrieving state from json_url: ${json_url}.`,
+                delay: true,
```
Comment (nkemnitz) [inline src/neuroglancer/viewer.ts]: Oh, haven't used `forPromise` before. Looks like the `delay` parameter is the time until the message *appears*, not the time until it *disappears* - which makes the message now completely invisible to the user :(
The `showTemporaryMessage` function has the more useful `closeAfter` parameter. Sorry 'bout that.

File: src/neuroglancer/viewer.ts [hunk 3]
```diff
@@ -617,50 +617,55 @@ export class Viewer extends RefCounted implements ViewerState {
...
+            initialMessage: `Posting state to: ${this.jsonStateServer.value}.`,
+            delay: true,
+            errorPrefix: `Error posting state: `,
           })
           // catch errors with upload and prompt the user if there was an error
-          .catch((err) => {
-            console.log('entered state server not responding:', this.jsonStateServer.value);
-            console.log(err);
+          .catch(() => {
+            console.log('not responding');
```
Comment (nkemnitz) [inline src/neuroglancer/viewer.ts]: Can remove console.log - prompt below contains the message

File: src/neuroglancer/viewer.ts (hunk 2 above)
Comment (fcollman) [reply src/neuroglancer/viewer.ts]: okay.. i think i've fixed this

Comment (fcollman) [reply src/neuroglancer/viewer.ts]: wait.. this one is good... it shows the error message when it errors and doesn't show much if the server responds quickly... which is what we want.
